### ISSUE TITLE
Search does not search 

### ISSUE THREAD

**Issue by leocrespo** [pojome/elementor#6473]
This search box is utterly useless. The search criteria is ignored altogether, results that don't contain the criteria are returned, there is no order whatsoever, it just returns everything in a random order.

This is a 100% reproducible bug, just try it out. Type anything in this box. 

<img width="292" alt="template search" src="https://user-images.githubusercontent.com/1483902/50058823-c369e380-0132-11e9-837f-3771de39d69a.png">

Developer "bainternet" closed this as a non issue saying that this is the correct behavior. Not sure in what planet a search box that ignores search criteria is the correct behavior.

**Comment by joshmarom** [comment]
The search mechanism used in this control, is WordPress' native search query functionality:
https://codex.wordpress.org/Class_Reference/WP_Query#Search_Parameter
Although it is admittedly not the most efficient search algorithm, it still is the standard way to query posts based on keyword in WordPress. 

The results are organized in groups according to their type:
Content, Section, Header, Footer, Single, Archive... etc...

@bainternet is right. There is no bug or issue here. In planet Earth, where people use WordPress, this is how things work.

**Comment by leocrespo** [comment]
Got a reply from support, they can reproduce this bug too, just like everyone else. Hopefully @bainternet and @joshmarom aren't the only ones looking into this. 
> Soyket repliedDec 18, 10:13Hi,I could reproduce this at my end. This has been reported as a bug to our dev team. Thank you for this.

**Comment by bainternet** [comment]
@leocrespo 	

Should be tweaked next version.

**Comment by black-eye** [comment]
I can confirm this issue, and I also mentioned it earlier here: https://github.com/pojome/elementor/issues/6271#issuecomment-442674572

Hopefully the issue will be really fixed and not just "tweaked"...

**Comment by leocrespo** [comment]
Thank you so much for the quick fix @bainternet.

**Comment by henrique-voni** [comment]
This is not fixed, i am experiencing the same problem on this website: www.sogimaepi.com.br 

Check the search widget in menu (magnifier icon), it only searches for specific content.

**Comment by razvanwaterford** [comment]
I hate this problem with Elementor Pro 2.6... the search just doesn't work

**Comment by vijayasanthi1993** [comment]
I'm also having the same issue

**Comment by primathon** [comment]
I have no idea _why_, but the template search functions (which internally use `WP_Query`) were bombing out when using a combination of 1) a custom post type (in this case, `elementor_library`) and 2) any search parameter (i.e. what you type in the box). After a half an hour of tinkering with it, I built a fix. Add this to your functions.php (or the Code Snippets plugin, which is way better):

```
function mx_fix_elementor_template_search($query)
{
    if ($query->is_search)
    {
        if ($query->query['post_type'] == 'elementor_library')
        {
            $query->set('post_type', ['elementor_library']);
        }
    }
    return $query;
}

add_filter('pre_get_posts', 'mx_fix_elementor_template_search');
```

Basically, if the query detects that it's trying to search for an `elementor_library` post type, it properly sets the query vars so that you actually get results back. Hope this helps someone else too!
